src/mscorlib/shared/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Takes a path to a symbolic link and attempts to place the link target path into the buffer. If the buffer is too
+        /// small, the path will be truncated. No matter what, the buffer will not be null terminated. 
+        /// </summary>
+        /// <param name="path">The path to the symlink</param>
+        /// <param name="buffer">The buffer to hold the output path</param>
+        /// <param name="bufferSize">The size of the buffer</param>
+        /// <returns>
+        /// Returns the number of bytes placed into the buffer on success; 0 if the buffer is too small; and -1 on error.
```

Comment:
I know this was copied from corefx - but `0 if the buffer is too small` is incorrect from reading [https://linux.die.net/man/2/readlink](https://linux.die.net/man/2/readlink)

Reply:
Oops - nevermind.  I see this code was updated an hour ago.  I was out of date.

---
In reply to: [153556312](https://github.com/dotnet/coreclr/pull/15229#discussion_r153556312) [](ancestors = 153556312)